javaparser-core/src/main/java/com/github/javaparser/ast/type/UnionType.java

```diff
@@ -32,6 +30,7 @@ public UnionType(NodeList<ReferenceType<?>> elements) {
     }
 
     public UnionType setElements(NodeList<ReferenceType<?>> elements) {
+        notifyPropertyChange("elements", this.elements, elements);
```

Comment:
right, I forgot to consider that users will use this. Ok, I can go with the enum.

If this is the only concern I will change every call to use the enum right now, if there are more I will apply all the changes at the same time.